demo/src/main/java/com/example/EventHub/Event/EventController.java

```diff
@@ -48,31 +48,16 @@ public class EventController {
     @Autowired
     UserRepository userRepository;
 
-    @GetMapping("/add")
-    public String addEvent(Model model) {
-        model.addAttribute("eventDTO", new EventDTO());
-        model.addAttribute("eventTypes", eventTypeRepository.findAll());
-        model.addAttribute("organisations", organisationRepository.findAll());
-        return "event-form";
-    }
-
     @PostMapping("/submit")
-    public String postEvent(@Valid @ModelAttribute EventDTO eventDTO, BindingResult bindingResult, Model model) throws ParseException {
+    public void postEvent(@Valid @ModelAttribute EventDTO eventDTO, BindingResult bindingResult)  {
         if (bindingResult.hasErrors()) {
-            model.addAttribute("eventTypes", eventTypeRepository.findAll());
-            model.addAttribute("organisations", organisationRepository.findAll());
-            return "event-form";
+            throw new IllegalArgumentException();
```

Comment:
I recommend adding some sort of text in the IllegalArgumentException(); so that if there is a problem with an api call, we will know directly the cause without that much looking.